core/src/main/scala/com/spotify/featran/FeatureSpec.scala

```diff
@@ -109,8 +109,62 @@ class FeatureSpec[T] private[featran] (private[featran] val features: Array[Feat
    * @param input input collection
    * @tparam M input collection type, e.g. `Array`, `List`
    */
-  def extract[M[_]: CollectionType](input: M[T]): FeatureExtractor[M, T] =
-    new FeatureExtractor[M, T](new FeatureSet[T](features, crossings), input, None)
+  def extract[M[_]: CollectionType](input: M[T]): FeatureExtractor[M, T] = {
+    val ct: CollectionType[M] = implicitly[CollectionType[M]]
+    import ct.Ops._
+
+    val fs = input.pure(new FeatureSet[T](features, crossings))
+    new FeatureExtractor[M, T](fs, input, None)
+  }
+
+  /**
+   * Extract features from an input collection based on the provided predicate
+   *
+   * @param input input collection
+   * @param predicate Function determining whether or not to include the feature
+   * @tparam M input collection type, e.g. `Array`, `List`
+   */
+  def extract[M[_]: CollectionType](
+    input: M[T],
+    predicate: Feature[T, _, _, _] => Boolean): FeatureExtractor[M, T] = {
```

Comment:
Curry `predicate` so one can do `.extract(in) { f => ... }`?

Reply:
@nevillelyh thought about that as well and then i was torn.. coz if we curry it we need to rename that method.

`extractWithPredicate`?

Reply:
Yeah that could work, or `extractSubset`?

Reply:
`extractSubset`